Configurations.md

```diff
@@ -2381,12 +2381,74 @@ Don't reformat out of line modules
 
 Maximum line length for single line if-else expressions. A value of `0` (zero) results in if-else expressions always being broken into multiple lines. Note this occurs when `use_small_heuristics` is set to `Off`.
 
+Common if-else expressions that may be formatted on a single line are:
+* Expressions on the right hand side of an equals (`=`) sign. e.g. `let x = if y == 1 { 0 } else { 1 };`
+* Conditional expression in function calls. e.g. `f(if x== 0 { 10 } else { 0 })`
+* Single expression closures. e.g. `let x = |y| if y == 0 { 0 } else { 10 };` 
+
+In the following scenarios the if-else expression will be formatted over multiple lines.
+* Empty if-else blocks.
+* An outer if-else expression when if-else expression are nested.
+* If-else blocks that contain statements (e.g. `let x = a;`), comments, or attributes (e.g. `let _ = if true { #[must_use] 1 } else { 2 };`).
+
+When `version=One` is set (the default) an if-else expression at the end of a block is treated as a statement and is not formatted on a single line. When using `version=Two` the if-else expression may be formatted on a single line if it meets the criteria described above.
+
```

Comment:
I have some reservations about any attempts to reference, or perhaps to some readers, codify formatting behavior here. The Style Guide is what articulates the rules for the formatting behavior, and the exact prescriptions for when `if-else` can/can't be single lined can be found [here](https://github.com/rust-lang/rust/blob/master/src/doc/style-guide/src/expressions.md#single-line-if-else).

This option exists as the mechanism to control the value used as the "small" threshold as described in the Guide, and I don't think we should have much more than that here.

Basically, the Config docs, and any particular option within them, isn't really the right place to try to explain the broader formatting rules for a given syntax construct